pkg/engine/planner/physical/optimizer.go

```diff
@@ -248,19 +277,86 @@ func (r *projectionPushdown) applyProjectionPushdown(node Node, projections []Co
 			}
 		}
 		return changed
+	case *ParseNode:
+		// Only apply the pushdown for Metric queries. Log queries should request all keys
+		if !r.isMetricQuery() || len(ambiguousProjections) == 0 {
+			return false
+		}
+
+		// Found a ParseNode - update its keys
+		existingKeys := make(map[string]bool)
+		for _, k := range node.RequestedKeys {
+			existingKeys[k] = true
+		}
+
+		// Add new keys
+		changed := false
+
+		for _, p := range ambiguousProjections {
+			colExpr, ok := p.(*ColumnExpr)
+			if !ok {
+				continue
+			}
+
+			// Only collect ambiguous columns to push to parse nodes
+			if colExpr.Ref.Type == types.ColumnTypeAmbiguous && !existingKeys[colExpr.Ref.Column] {
+				existingKeys[colExpr.Ref.Column] = true
+				changed = true
+			}
```

Comment:
nit: this can be simplified a bit to just setting `existingKeys[colExpr.Ref.Column] = true`, since the projections are always ambiguous. Then, after the loop, you can initialize changed to `changed = len(existingKeys) > len(node.RequestedKeys)`.